### PR TITLE
Added the countries attribute and check_iso3_codes() method to RegionCode object

### PR DESCRIPTION
**Pull request by GretchenSchowalter**
Hello! This was my initial idea in tackling issue #224. Currently, the check_iso3_codes() method returns a Boolean, but we can change that if there is a better idea in how to do this!

### CODE REVIEW

File: tests/test_code.py [hunk 1]
```diff
@@ -55,3 +56,27 @@ def test_RegionCode_hierarchy_attribute():
     )
 
     assert reg.hierarchy == "R5"
+
+
+def test_RegionCode_iso3_code():
+    reg = RegionCode(
+        name="OECD",
```
Comment (danielhuppmann) [inline tests/test_code.py:63]: I suggest to use a more "correct" test, e.g. Western Europe that has France, Germany, Spain, Portugal...

File: nomenclature/code.py [hunk 2]
```diff
@@ -209,3 +211,11 @@ class RegionCode(Code):
     """
 
     hierarchy: str = None
+    countries: List[str] = None
+
+    def check_iso3_codes(self) -> bool:
+        for country in self.countries:
+            if pycountry.countries.get(alpha_3=country) is None:
+                logging.warning(f"{country} is not a valid iso3 country code!")
```
Comment (danielhuppmann) [inline nomenclature/code.py:219]: Better to keep a list all invalid countries in this check, and return a warning like:
```python
f"Region {name} has invalid ISO3 country codes: {invalid_iso3_codes}"
```

**Review by danielhuppmann — COMMENTED** [review]
Looks good, two minor comments.

Good question whether to simply write a warning or do a "hard" ValueError - any preference, @phackstock?

**Comment by danielhuppmann** [comment]
And you'll have to add the pycountry package to the dependencies...

**Comment by phackstock** [comment]
> Good question whether to simply write a warning or do a "hard" ValueError - any preference, @phackstock?

@danielhuppmann hm, I could see arguments for both but my preference is always to fail as soon and as loudly as possible. So I'd vote error.

File: nomenclature/code.py (hunk 2 above)
Comment (phackstock) [reply nomenclature/code.py:219]: That would be good to turn into a pydantic validator. Reason being is that it is then automatically called whenever you create an instance of `RegionCode`. For now it has to be called explicitly.

File: tests/test_code.py [hunk 3]
```diff
@@ -55,3 +56,27 @@ def test_RegionCode_hierarchy_attribute():
...
+    assert reg.check_iso3_codes() is True
+
+
+def test_RegionCode_iso3_code_fail(caplog):
+    reg = RegionCode(
+        name="OECD",
+        hierarchy="R5",
+        countries=["ALB", "BGR", "DEX", "AUS"],
+    )
+    caplog.set_level(logging.WARNING)
```
Comment (phackstock) [inline tests/test_code.py:77]: Do you need this and the following line?

**Review by phackstock — CHANGES_REQUESTED** [review]
Looks almost ready to me. 
One suggestion below regarding the iso3 validator.

**Comment by GretchenSchowalter** [comment]
Hello! I've been having a little trouble figuring out how to implement the @validator to the method. Currently the test_RegionCode_iso3_codes_fail() is still failing, so if you have any suggestions that would be great!

File: nomenclature/code.py [hunk 4]
```diff
@@ -209,3 +210,17 @@ class RegionCode(Code):
...
+    countries: List[str] = None
+
+    @validator("countries")
+    def check_iso3_codes(cls, v, values) -> bool:
+        invalid_iso3_codes: List[str] = []
+        for country in v:
+            if pycountry.countries.get(alpha_3=country) is None:
+                invalid_iso3_codes.append(country)
+        if invalid_iso3_codes:
+            raise ValidationError(
```
Comment (phackstock) [inline nomenclature/code.py:222]: This needs to be a `ValueError` (or any kind of other error except `pydantic.ValidationError`:
```suggestion
            raise ValueError(
```
Reason being, pydantic collects all the errors found during validation and raises them as one common error which is a `pydantic.ValidationError`. The individual errors, however, cannot be `pydantic.ValidationError`.

File: nomenclature/code.py [hunk 5]
```diff
@@ -209,3 +210,17 @@ class RegionCode(Code):
...
+                f"Region {values['name']} has invalid\n"
+                " ISO3 country codes: {invalid_iso3_codes}"
```
Comment (phackstock) [inline nomenclature/code.py:224]: You'll need another f here to make string interpolation work
```suggestion
                f" ISO3 country codes: {invalid_iso3_codes}"
```

**Review by phackstock — APPROVED** [review]
Looks good to me thanks @GretchenSchowalter.
Good to be merged form my side.

**Review by danielhuppmann — APPROVED** [review]
Nice work, thank you!

**Pull request merged**